azurerm/internal/services/mssqlvm/resource_arm_mssql_virtual_machine.go

```diff
@@ -0,0 +1,559 @@
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/features"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmMsSqlVirtualMachine() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmMsSqlVirtualMachineCreateUpdate,
+		Read:   resourceArmMsSqlVirtualMachineRead,
+		Update: resourceArmMsSqlVirtualMachineCreateUpdate,
+		Delete: resourceArmMsSqlVirtualMachineDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"location": azure.SchemaLocation(),
+
+			"resource_group_name": azure.SchemaResourceGroupNameDiffSuppress(),
+
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"virtual_machine_resource_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
+					v := val.(string)
+					_, err := azure.ParseAzureResourceID(v)
+					errs = append(errs, err)
+					return
+				},
+			},
+
+			"sql_server_license_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(sqlvirtualmachine.PAYG),
+					string(sqlvirtualmachine.AHUB),
+				}, false),
+			},
+
+			"sql_image_sku": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(sqlvirtualmachine.Developer),
+					string(sqlvirtualmachine.Express),
+					string(sqlvirtualmachine.Standard),
+					string(sqlvirtualmachine.Enterprise),
+					string(sqlvirtualmachine.Web),
+				}, false),
+			},
+
+			"sql_virtual_machine_group_resource_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
+					v := val.(string)
+					_, err := azure.ParseAzureResourceID(v)
+					errs = append(errs, err)
+					return
+				},
+			},
+
+			"auto_patching_settings": {
```

Comment:
We don't need to append settings here, it is implied
```suggestion
			"auto_patching": {
```